### ISSUE TITLE
GNS3 server 1.5.2 ; AttributeError: module 'aiohttp.web' has no attribute 'RequestHandler'

### ISSUE THREAD

**Issue by AODtorusan** [GNS3/gns3-server#842]
I upgraded gns3-server via pip3, but now it fails to properly start;

```
root@CPE-1:~# pip3 install gns3-server 
Requirement already satisfied (use --upgrade to upgrade): gns3-server in /usr/local/lib/python3.5/dist-packages
Requirement already satisfied (use --upgrade to upgrade): Jinja2>=2.7.3 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): raven>=5.2.0 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): aiohttp>=0.21.5 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): zipstream>=1.1.3 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): jsonschema>=2.4.0 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): psutil>=3.0.0 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already satisfied (use --upgrade to upgrade): MarkupSafe in /usr/local/lib/python3.5/dist-packages (from Jinja2>=2.7.3->gns3-server)
Requirement already satisfied (use --upgrade to upgrade): contextlib2 in /usr/local/lib/python3.5/dist-packages (from raven>=5.2.0->gns3-server)
Requirement already satisfied (use --upgrade to upgrade): chardet in /usr/local/lib/python3.5/dist-packages (from aiohttp>=0.21.5->gns3-server)
Requirement already satisfied (use --upgrade to upgrade): multidict>=2.0 in /usr/local/lib/python3.5/dist-packages (from aiohttp>=0.21.5->gns3-server)
Requirement already satisfied (use --upgrade to upgrade): async-timeout in /usr/local/lib/python3.5/dist-packages (from aiohttp>=0.21.5->gns3-server)
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
root@CPE-1:~# pip3 install gns3-server  --^C
root@CPE-1:~# pip install --upgrade pip
-bash: pip: command not found
root@CPE-1:~# pip3 install --upgrade pip
Collecting pip
  Downloading pip-9.0.1-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 758kB/s 
Installing collected packages: pip
  Found existing installation: pip 8.1.2
    Not uninstalling pip at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed pip-9.0.1
root@CPE-1:~# pip3 install gns3-server --upgrade
Requirement already up-to-date: gns3-server in /usr/local/lib/python3.5/dist-packages
Collecting aiohttp>=0.21.5 (from gns3-server)
  Downloading aiohttp-1.2.0-cp35-cp35m-manylinux1_x86_64.whl (155kB)
    100% |████████████████████████████████| 163kB 2.4MB/s 
Requirement already up-to-date: Jinja2>=2.7.3 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Requirement already up-to-date: zipstream>=1.1.3 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Collecting psutil>=3.0.0 (from gns3-server)
  Downloading psutil-5.0.0.zip (374kB)
    100% |████████████████████████████████| 378kB 1.8MB/s 
Collecting raven>=5.2.0 (from gns3-server)
  Downloading raven-5.32.0-py2.py3-none-any.whl (277kB)
    100% |████████████████████████████████| 286kB 2.2MB/s 
Requirement already up-to-date: jsonschema>=2.4.0 in /usr/local/lib/python3.5/dist-packages (from gns3-server)
Collecting multidict>=2.1.4 (from aiohttp>=0.21.5->gns3-server)
  Downloading multidict-2.1.4-cp35-cp35m-manylinux1_x86_64.whl (357kB)
    100% |████████████████████████████████| 358kB 1.8MB/s 
Requirement already up-to-date: chardet in /usr/local/lib/python3.5/dist-packages (from aiohttp>=0.21.5->gns3-server)
Collecting yarl>=0.8.1 (from aiohttp>=0.21.5->gns3-server)
  Downloading yarl-0.8.1-cp35-cp35m-manylinux1_x86_64.whl (137kB)
    100% |████████████████████████████████| 143kB 3.2MB/s 
Requirement already up-to-date: async-timeout>=1.1.0 in /usr/local/lib/python3.5/dist-packages (from aiohttp>=0.21.5->gns3-server)
Requirement already up-to-date: MarkupSafe in /usr/local/lib/python3.5/dist-packages (from Jinja2>=2.7.3->gns3-server)
Requirement already up-to-date: contextlib2 in /usr/local/lib/python3.5/dist-packages (from raven>=5.2.0->gns3-server)
Building wheels for collected packages: psutil
  Running setup.py bdist_wheel for psutil ... done
  Stored in directory: /root/.cache/pip/wheels/67/5c/c3/58a3b52db99b3897a190788d4b6ba7dfef44cff5b3b594cbf1
Successfully built psutil
Installing collected packages: multidict, yarl, aiohttp, psutil, raven
  Found existing installation: multidict 2.1.2
    Uninstalling multidict-2.1.2:
      Successfully uninstalled multidict-2.1.2
  Found existing installation: aiohttp 1.0.5
    Uninstalling aiohttp-1.0.5:
      Successfully uninstalled aiohttp-1.0.5
  Found existing installation: psutil 4.4.0
    Uninstalling psutil-4.4.0:
      Successfully uninstalled psutil-4.4.0
  Found existing installation: raven 5.31.0
    Uninstalling raven-5.31.0:
      Successfully uninstalled raven-5.31.0
Successfully installed aiohttp-1.2.0 multidict-2.1.4 psutil-5.0.0 raven-5.32.0 yarl-0.8.1
root@CPE-1:~# pip3 list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
aiohttp (1.2.0)
async-timeout (1.1.0)
chardet (2.3.0)
contextlib2 (0.5.4)
gns3-server (1.5.2)
Jinja2 (2.8)
jsonschema (2.5.1)
MarkupSafe (0.23)
multidict (2.1.4)
pip (9.0.1)
psutil (5.0.0)
raven (5.32.0)
setuptools (28.0.0)
wheel (0.29.0)
yarl (0.8.1)
zipstream (1.1.4)
root@CPE-1:~# gns3server 
Traceback (most recent call last):
  File "/usr/local/bin/gns3server", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/gns3server/main.py", line 71, in main
    from gns3server.run import run
  File "/usr/local/lib/python3.5/dist-packages/gns3server/run.py", line 32, in <module>
    from gns3server.server import Server
  File "/usr/local/lib/python3.5/dist-packages/gns3server/server.py", line 33, in <module>
    from .web.request_handler import RequestHandler
  File "/usr/local/lib/python3.5/dist-packages/gns3server/web/request_handler.py", line 22, in <module>
    class RequestHandler(aiohttp.web.RequestHandler):
AttributeError: module 'aiohttp.web' has no attribute 'RequestHandler'
```

Is the dependency on aiohttp the correct version ?

**Comment by noplay** [comment]
pip3 install aiohttp==1.1.6 will fix the issue for you

@grossmj what we need to do ? We could backport this 572fa08049eb53da9cb27d5f8b1af0f6ed7e4ed8 and 3bc78f61df3f3c3364fe0aa570f683bd232111d2 or lock the aiohttp version to < 1.2

But how to release it? Installation via pip is broken, but due to PyPi limitation we need to increment version number to upload a new version. If we release a 1.5.3 just on PyPi people will want an 1.5.3 for Windows, OSX, Ubuntu...

The situation is not easy to avoid, we can't lock a version without creating trouble for the rolling release linux distribution like Arch Linux.

**Comment by grossmj** [comment]
Not an easy one. I would backport the commits and release 1.5.3 for all platforms.

**Comment by AODtorusan** [comment]
Just to confirm, forcing aiohttp v1.1.6 fixes the issue.

**Comment by itsakademiks** [comment]
Is this accepted as the answer? i too am having this same issue. after i install aiohttp v1.1.6 i get a successful completion. after i do a pip3 list it still shows version 1.2.0


[dom@HOST1 Downloads]$ pip3 install --user aiohttp==1.1.6
Collecting aiohttp==1.1.6
  Using cached aiohttp-1.1.6-cp35-cp35m-manylinux1_x86_64.whl
Requirement already satisfied: yarl>=0.5.0 in /usr/local/lib/python3.5/site-packages/yarl-0.8.1-py3.5-linux-x86_64.egg (from aiohttp==1.1.6)
Requirement already satisfied: chardet in /usr/local/lib/python3.5/site-packages/chardet-2.3.0-py3.5.egg (from aiohttp==1.1.6)
Requirement already satisfied: multidict>=2.0 in /usr/local/lib/python3.5/site-packages/multidict-2.1.4-py3.5-linux-x86_64.egg (from aiohttp==1.1.6)
Requirement already satisfied: async-timeout>=1.1.0 in /usr/local/lib/python3.5/site-packages/async_timeout-1.1.0-py3.5.egg (from aiohttp==1.1.6)
Installing collected packages: aiohttp
Successfully installed aiohttp-1.1.6
[dom@HOST1 Downloads]$ pip3 list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
aiohttp (1.2.0)
async-timeout (1.1.0)
chardet (2.3.0)
configobj (5.0.6)
gns3-gui (1.4.4)
gns3-net-converter (1.3.0)
gns3-server (1.4.4)
Jinja2 (2.8.1)
jsonschema (2.5.1)
MarkupSafe (0.23)
multidict (2.1.4)
netifaces (0.10.4)
pip (9.0.1)
psutil (5.0.1)
raven (5.32.0)
setuptools (12.2)
six (1.10.0)
virtualenv (15.1.0)
ws4py (0.3.5)
yarl (0.8.1)
zipstream (1.1.4)

**Comment by noplay** [comment]
If you use --user you will install the module just for you but if the module is aiothttp is installed for everyone Python will use it.

**Comment by itsakademiks** [comment]
noplay so you're saying run the command as "pip3 install aiohttp==1.1.6"? 

when i do that i get permission denied thats why i used the --user switch.

**Comment by noplay** [comment]
sudo pip3 install aiohttp==1.1.6 should work

What is your linux version?

**Comment by itsakademiks** [comment]
centos 7. i'm at work but have ssh access into that box I was able to
su
then
pip3 install aiohttp==1.1.16

i then ran gns3server i didnt see any errors 
also tried gns3 but i dont have x11 forwarding turned on so i wasnt able to see if it is successful
will be back with more info one i get in front of the box

**Comment by noplay** [comment]
Solved by the 1.5.3

**Comment by IsraelCecil** [comment]
Had the same issue. Doing **pip3 install aiohttp==1.1.6** resolved the problem for me.
